src/query/service/src/sessions/query_ctx.rs

```diff
@@ -108,7 +108,7 @@ pub struct QueryContext {
     query_settings: Arc<Settings>,
     fragment_id: Arc<AtomicUsize>,
     // Used by synchronized generate aggregating indexes when new data written.
-    inserted_segment_locs: Arc<RwLock<Vec<Location>>>,
+    inserted_segment_locs: Arc<RwLock<HashSet<Location>>>,
```

Comment:
Why change it to `HashSet`? There will be same `Location`s?

Reply:
If multiple operations (eg: insert, update) in one query ctx, there will maybe be add same location to query context in different processor.